### PR TITLE
[WIP] P-eagle Training Phase 1 implementation

### PR DESCRIPTION
**Pull request by shanjiaz**
PLEASE FILL IN THE PR DESCRIPTION HERE ENSURING ALL CHECKLIST ITEMS (AT THE BOTTOM) HAVE BEEN CONSIDERED.

## Tests

I have filled in:

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan/results, such as providing test command and pasting the results.
- [ ] (Optional) The necessary documentation update.
- [x] I (a human) have written or reviewed the code in this pr to the best of my ability.

### CODE REVIEW

**Comment by ivanbaldo** [comment]
Sorry for my ignorance, but I wonder if dflash isn't a better alternative to P-eagle and if P-eagle is worth to pursuing nowadays...?

File: examples/data_generation_and_training/peagle_qwen3_8b_example.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        max_samples=500000,
+    )
+
+    # Vocab Mapping - set to None to use full vocabulary (no reduction)
+    vocab_mapping_args = None
+
+    train_args = TrainArgs(
+        logger="trackio",
+        lr=1e-4,
+        total_seq_len=TOTAL_SEQ_LEN,
```
Comment (Neo9061) [inline examples/data_generation_and_training/peagle_qwen3_8b_example.py:40]: what does this parameter mean? total effective sequence length after multiplied by k groups?

File: examples/data_generation_and_training/peagle_llama3_8b_example.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        total_seq_len=TOTAL_SEQ_LEN,
+        run_name="peagle_llama3_8b_sharegpt_5k",
+        epochs=10,
+        num_layers=4,
+        speculator_type="peagle",  # Use P-EAGLE instead of EAGLE-3
+        para_depths=10,
+        down_sample_ratio=0.7,
+        down_sample_ratio_min=0.2,
+        max_seq_len=2048,
+        ptd_token_id=128255,
```
Comment (Neo9061) [inline examples/data_generation_and_training/peagle_llama3_8b_example.py:48]: we should use something else as mask id, 128255 has already been used in llama 3.1 8B config.
```
"128255": {
--
"content": "<\|reserved_special_token_247\|>",
"lstrip": false,
"normalized": false,
"rstrip": false,
"single_word": false,
"special": true
}
```

File: src/speculators/models/peagle/core.py [hunk 3]
```diff
@@ -0,0 +1,450 @@
...
+        self.down_sample_ratio = config.down_sample_ratio
+        self.down_sample_ratio_min = config.down_sample_ratio_min
+        self.ptd_token_id = config.ptd_token_id
+        self.max_seq_len = config.max_seq_len
+
+        self.lm_head.weight.requires_grad = False
+        self.lm_head.eval()
+
+        # Learnable mask_hidden parameter for padding unsampled positions
+        self.mask_hidden = torch.nn.Parameter(torch.zeros(1, 1, 3 * self.hidden_size))
```
Comment (Neo9061) [inline src/speculators/models/peagle/core.py:66]: might be worthy to use `torch.randn` to init the mask_hidden for faster convergency.

File: examples/data_generation_and_training/peagle_llama3_8b_example.py (hunk 2 above)
Comment (Neo9061) [reply examples/data_generation_and_training/peagle_llama3_8b_example.py:48]: I used `128020` for ptd_id
